htmd/molecule/molecule.py

```diff
@@ -310,14 +313,17 @@ def insertappend(index, data1, data2, append):
                     self.bonds = newbonds
                     self.bondtype = mol.bondtype
 
-            for k in self._atom_fields:
+            for k in self._atom_and_coord_fields:
                 if k == 'serial':
                     continue
                 data2 = mol.__dict__[k]
                 if mol.__dict__[k] is None or np.size(mol.__dict__[k]) == 0:
                     data2 = self._empty(mol.numAtoms, k)
                 self.__dict__[k] = insertappend(index, self.__dict__[k], data2, append)
             self.serial = np.arange(1, self.numAtoms + 1)
+            # Reset the box to zeros as you cannot keep box size after inserting atoms
```

Comment:
Why not? The user could have setup a particular box size and be using some protocol to insert molecules inside it (imagine some solvate procedure with organic molecules). Or am I missing the point here?